File: code/strings/string-interpolation.erl
% Erlang doesn't have direct string interpolation
% Use io_lib:format/2 or similar
-module(example).
-export([start/0]).

start() -&gt;
    Apples = 3,
    Oranges = 5,
    FruitSummary = io_lib:format("I have ~p ~s", [Apples + Oranges, "pieces of fruit."]),
    io:format("~s~n", [FruitSummary]).

File: code/strings/string-interpolation.m
#import &lt;Foundation/Foundation.h&gt;

int main() {
    int apples = 3;
    int oranges = 5;
    NSString *fruitSummary = [NSString stringWithFormat:@"I have %d %@",
                              apples + oranges,
                              @"pieces of fruit."];
    NSLog(@"%@", fruitSummary);
    return 0;
}